Gemfile

```diff
@@ -19,6 +19,7 @@ gem "pagy", "~> 5.10"
 gem "puma", "~> 5.0"
 gem "rails", "~> 6.1.7"
 gem "rails-i18n"
+gem "ransack"
```

Comment:
gem ransack nó hỗ trợ search ntn e?
câu query truy vấn ứng tới từng predicate phổ biến là gì

Reply:
# gem ransack cung cấp các method hỗ trợ cho việc tạo các câu query truy vấn đến database#
# câu query truy vấn ứng với từng predicate phổ biến:
- eq: query sẽ trả về tất cả các bảng record mà field bằng chính xác giá trị đã cho, tương ứng với dấu "=" trong where
- matches: query sẽ trả về các record mà field có giá trị giống giá trị đã cho, tương ứng với "LIKE" trong where
- lt: query sẽ trả về các record mà field có giá trị nhỏ hơn giá trị đã cho, tương ứng với dấu "<" trong where
- lteq: query sẽ trả về các record mà field có giá trị nhỏ hơn hoặc bằng giá trị đã cho, tương ứng với dấu "<=" trong where
- gt: query sẽ trả về các record mà field có giá trị lớn hơn giá trị đã cho, tương ứng với dấu ">" trong where
- gteq: query sẽ trả về các record mà field có giá trị lớn hơn hoặc bằng giá trị đã cho, tương ứng với dấu ">=" trong where 
- in: query trả về các record mà field có giá trị thuộc về tập các giá trị đã cho, tương ứng với "IN" trong where 
- cont: query trả về các record mà field có giá trị chứa giá trị đã cho, tương ứng với "LIKE '%value%' " trong where 
- start: query trả về các record mà field có giá trị bắt đầu bằng giá trị đã cho, tương ứng với "LIKE 'value%' " trong where 
- end: query trả về các record mà field có giá trị kết thúc bằng giá trị đã cho, tương ứng với "LIKE '%value' " trong where 
- true: query trả về các record mà field có giá trị đúng.
- flase: query trả về các record mà field có giá trị sai.